### PR TITLE
WIP: room invite redesign

### PR DESCRIPTION
**Pull request by suprko**
#### Relevant info and issue/PR links 
https://app.clubhouse.io/peerio/story/273/desktop-room-invites-redesigned-flow
https://github.com/PeerioTechnologies/peerio-icebear/pull/141

#### Testing instructions  
This is what works so far:
* Room invites are listed along with chats, in alphabetical order, according to mockups
* Clicking room invite opens `ChannelInvite` component
* Host avatar is shown
* Accepting and rejecting invite works with existing SDK
  * very minor modifications to SDK so far; just made channel activate after accepting invite, which is how it used to work

SDK fixes (future):
* Need to be able to mark invites as `active` in SDK, the way rooms/DMs can be marked as `active`
  * Selecting a room invite does not currently set its entry in `ChatList` to `active` style, i.e. white text on blue background
  * ~~After declining an invite, selecting another room invite still shows the "invite declined" screen, because all room invites share the same component (will fix on client side for now)~~
* Similarly, need to be able to mark invites as `read`/`unread` in SDK
  * Selecting a room invite does not currently remove the `unread` style, i.e. semibold text
* No list of participants other than host

----
### Repository owner

- [ ] Was tested and can be merged.
- [ ] PR name follows conventional changelog format.

[PR guideline](https://github.com/PeerioTechnologies/peerio-icebear/blob/dev/docs/CONTRIBUTING.md)

### CODE REVIEW

File: src/ui/chat/components/ChatNameEditor.jsx [hunk 1]
```diff
@@ -13,7 +13,7 @@ class ChatNameEditor extends React.Component {
     }
 
     componentDidMount() {
-        this.inputRef.focus();
+        if (this.inputRef) this.inputRef.focus();
```
Comment (seavan) [inline src/ui/chat/components/ChatNameEditor.jsx]: @suprko this should be fixed in #dev, right?

Comment (suprko) [reply src/ui/chat/components/ChatNameEditor.jsx]: Yeah I've already made a PR https://github.com/PeerioTechnologies/peerio-desktop/pull/236

**Comment by suprko** [comment]
@seavan so the issues I've listed in "Testing instructions" -- should I fix those on UI side? or wait for SDK changes? I don't think I can do the SDK part myself.

**Comment by seavan** [comment]
please create separate stories for them, these are SDK and out of scope currently @suprko

**Comment by suprko** [comment]
@seavan OK but a couple of them are pretty serious. e.g. not being able to open another room invite from a "invite declined" screen. We can't ship with that, so if SDK is out of scope then I'll need to do a fix on client side

**Comment by seavan** [comment]
invite declined I'll fix, it's client thing, not sdk. please leave it here for me as a note.

**Comment by suprko** [comment]
@seavan Eren requesting review for the new changes before testing

**Pull request closed without merging**